syntax = "proto3";

package sdkproxy;

option cc_generic_services = true;

message HttpRequest { };
message HttpResponse { };

/////////////////////////////////////////////////////////////////////////////////////////////////////
// 设备查询服务
/////////////////////////////////////////////////////////////////////////////////////////////////////
service DeviceQueryService{
	rpc Discovery(HttpRequest) returns (HttpResponse) {};
	rpc Query(HttpRequest) returns (HttpResponse) {};
}

/////////////////////////////////////////////////////////////////////////////////////////////////////
// 实时流服务
/////////////////////////////////////////////////////////////////////////////////////////////////////
service RealStreamService{
    rpc Start(HttpRequest) returns (HttpResponse) {}
}

/////////////////////////////////////////////////////////////////////////////////////////////////////
// 点播服务
/////////////////////////////////////////////////////////////////////////////////////////////////////
service VodService{
	rpc Query(HttpRequest) returns (HttpResponse) {};
	rpc DownloadByTime(HttpRequest) returns (HttpResponse) {};
}

/////////////////////////////////////////////////////////////////////////////////////////////////////
// 事件分析服务
/////////////////////////////////////////////////////////////////////////////////////////////////////
service EventAnalyzeService{
	rpc Reset(HttpRequest) returns (HttpResponse) {}
	rpc Query(HttpRequest) returns (HttpResponse) {}
	rpc Start(HttpRequest) returns (HttpResponse) {}
	rpc Stop(HttpRequest) returns (HttpResponse) {}
}

/////////////////////////////////////////////////////////////////////////////////////////////////////
// 健康检查服务
/////////////////////////////////////////////////////////////////////////////////////////////////////
service HealthService{
	rpc Health(HttpRequest) returns (HttpResponse) {}
}

/////////////////////////////////////////////////////////////////////////////////////////////////////
// 系统配置服务
/////////////////////////////////////////////////////////////////////////////////////////////////////
service ConfigService{
	rpc GetFtp(HttpRequest) returns (HttpResponse) {}
	rpc SetFtp(HttpRequest) returns (HttpResponse) {}
}

/////////////////////////////////////////////////////////////////////////////////////////////////////
// 客流量服务
/////////////////////////////////////////////////////////////////////////////////////////////////////
service VisitorsFlowRateService{
	rpc QueryHistory(HttpRequest) returns (HttpResponse) {}
}

/////////////////////////////////////////////////////////////////////////////////////////////////////
// 门禁机服务
/////////////////////////////////////////////////////////////////////////////////////////////////////
service EntranceGuardService{
	 
}